### ISSUE TITLE
Rollover doesn't occur with more complicated date-formats

### ISSUE THREAD

**Issue by AndrewAshling** [lumoe/obsidian-rollover-daily-todos#56]
Rollover works fine when used with a simple date-format in the core plugin Daily Notes, e.g. YYYY-MM-DD

However, Daily Notes allows the user to apply more complicated formats like YYYY/MM/YYYY-MM-DD, which automatically places the daily note in a folder of the current year, and a subfolder of the current month. In this case the plugin doesn't work.

So the choice is between copying the non-finished tasks manually ***or*** changing the subfolder manually each month.

Is there a way to have both?

**Comment by lumoe** [comment]
Agreed, I do have the same issue, having this option would be awesome!

I think I was able to pinpoint the problem to this comparison: 

https://github.com/shichongrui/obsidian-rollover-daily-todos/blob/04204014f252cc8bf04ba66d756459dadb4ae72e/src/index.js#L121

When using this _special_ date format, the variables have the following state:
```
todayFormatted = 2022/Oct/2022-10-02
file.basename = 2022-10-02
```

I am new to Obsidian Plugin development, but my first observation was that there is no state where the current file is not the not today's daily file, which means that one could remove this comparison. I tested this by commenting it out in the dev tools, and it worked. 

If the comparison is required, one has to use the full file path, remove the daily folder and file extension and then do the comparison with the formatted date. 

I am happy to create a pull request for whichever case makes more sense if the author still supports this plugin.

**Comment by ovo-saxon-zerbino** [comment]
As that PR got lost when the repo changed hands I've re-raised it here https://github.com/lumoe/obsidian-rollover-daily-todos/pull/61

**Comment by lumoe** [comment]
Thank you for re-submitting the PR and contributing your solution to this long existing feature request @ovo-saxon-zerbino!

I have streamlined the changes into a new branch ([obsidian-rollover-daily-todos/support_directories](https://github.com/lumoe/obsidian-rollover-daily-todos/tree/support_directories)) and fixed an additional bug from #57 and added your changes. I would like to invite you for code reviews and testing it out.

**Comment by ovo-saxon-zerbino** [comment]
I should be able to take a look tomorrow, thanks @lumoe
